python/across_api/base/fov.py

```diff
@@ -0,0 +1,351 @@
+
+
+class FootprintFOV(FOVBase):
+    """
+    Constrained instrumet FOV. This is an FOV that calculate lates what is
+    visible in the FOV at a given pointing
+    """
+
+    footprint: Footprint
+
+    def __init__(self, pointing: PointingBase) -> None:
+        projected_footprint = self.footprint.project(
+            ra=pointing.ra, dec=pointing.dec, pos_angle=pointing.position_angle
+        )
+
+        ras_poly = [x[0] for x in projected_footprint][:-1]
+        decs_poly = [x[1] for x in projected_footprint][:-1]
+
+        cartesian_vertices = spherical_to_cartesian(
+            1,  # radial component
+            np.deg2rad(decs_poly),  # LAT - DECs
+            np.deg2rad(ras_poly),  # LON - RAs
+        )
+
+        self.visible_pixels = hp.query_polygon(
+            hp.order2nside(HEALPIX_MAP_EVAL_ORDER),
+            np.array(cartesian_vertices).T,
+            inclusive=True,
+            nest=True,
+        )
+
+
+class AllSkyFOV(FOVBase):
+    """
+    All sky instrument FOV. This is a simple FOV that is always visible unless
+    Earth occulted.
+    """
+
+    earth_constraint: EarthLimbConstraint
+
+    def __init__(self, ephem: EphemBase, time: Time):
+        """
+        Finds all healpix pixels within the ephemeris time slot that are not earth
+        occulted, and sets them to FOVBase.visible_pixels.
+
+        This is done by finding the antipodal position of the earth and calculating
+        all of the pixels within a disc of radius 180deg - radius_of_earth(deg)
+
+        Antipodal:
+            a_ra = ra-180 (deg)
+            a_dec = -dec  (deg)
+            a_radius = 180-radius (deg)
+
+        Function does assume that it is a space mission with available ephemeris
+        """
+        n_side = hp.order2nside(HEALPIX_MAP_EVAL_ORDER)
+        i_slice = get_slice(time=time, ephem=ephem)
+
+        # antipodal positions
+        ra = ephem.earth.ra[i_slice].degree - 180.0
+        dec = -1.0 * ephem.earth.dec[i_slice].degree
+
+        vec = hp.ang2vec(ra, dec, lonlat=True)
+
+        # np.array to store visible pixels
+        visible_pixels = np.array([]).astype(int)
+
+        # loop over each time-slice vector and concatenate
+        # visible pixels calculated with healpy's cone search (query_disc)
+        # assume the earth is a disc at ephem.earth position with radius
+        # radius = 180 - earthsize + constraint min angle.
+        for i, v in enumerate(vec):
+            radius = (
+                np.radians(180)
+                - np.radians(ephem.earthsize[i].value)
+                + np.radians(self.earth_constraint.min_angle.value)
+            )
+            visible_pixels = np.concatenate(
+                (
+                    visible_pixels,
+                    hp.query_disc(n_side, v, radius, nest=True, inclusive=False),
+                )
+            )
+
+        # only need unique pixels
+        self.visible_pixels = np.unique(visible_pixels)
```

Comment:
You may be able to do this even faster using query_polygon. query_polygon is implemented by calling query_disc for every edge, then taking the intersection. To employ query_disc to take the union, employ de Morgan's Law.

Reply:
Scratch that; these are not great circles, so `query_polygon` wouldn't work.